e2e/cypress/integration/enterprise/system_console/download_compliance_export_file_spec.js

```diff
@@ -0,0 +1,202 @@
+        exportCompliance();
+
+        // Get the download link
+        cy.get('@firstRow').findByText('Download').parents('a').should('exist').then((fileAttachment) => {
+            const fileURL = fileAttachment.attr('href');
+
+            // Download the File
+            downloadAttachmentAndVerifyItsProperties(fileURL);
+
+            // Export compliance again
+            exportCompliance();
+
+            // Download link should not exist this time
+            cy.get('.job-table__table').
+                find('tbody > tr:eq(0)').
+                findByText('Download').should('not.exist');
+        });
+    });
+
+    it('MM-T3439 - Download Compliance Export Files - S3 Bucket Storage', () => {
+        // Goto file storage settings Page
+        cy.visit('/admin_console/environment/file_storage');
+
+        // Get AWS credentials
+        const AWS_S3_BUCKET = Cypress.env('AWS_S3_BUCKET');
+        const AWS_ACCESS_KEY_ID = Cypress.env('AWS_ACCESS_KEY_ID');
+        const AWS_SECRET_ACCESS_KEY = Cypress.env('AWS_SECRET_ACCESS_KEY');
+
+        // Config AWS settings
+        cy.get('select[data-testid="FileSettings.DriverNamedropdown"]').select('amazons3');
+        cy.get('input[data-testid="FileSettings.AmazonS3Bucketinput"]').type(AWS_S3_BUCKET);
+        cy.get('input[data-testid="FileSettings.AmazonS3AccessKeyIdinput"]').type(AWS_ACCESS_KEY_ID);
+        cy.get('input[data-testid="FileSettings.AmazonS3SecretAccessKeyinput"]').type(AWS_SECRET_ACCESS_KEY);
+
+        // Save file storage settings
+        cy.get('button[data-testid="saveSetting"]').click();
+
+        waitUntilConfigSave();
+
+        // Goto compliance page and enable export
+        gotoCompliancePage();
+        enableComplianceExport();
+
+        // Navigate to a team and post an attachment
+        gotoTeamAndPostImage();
+
+        // Goto compliance page and start export
+        gotoCompliancePage();
+        exportCompliance();
+
+        // Get the download link
+        cy.get('@firstRow').findByText('Download').parents('a').should('exist').then((fileAttachment) => {
+            const fileURL = fileAttachment.attr('href');
+
+            // Download the file
+            downloadAttachmentAndVerifyItsProperties(fileURL);
+        });
+    });
+});
+
+function enableComplianceExport() {
+    // Enable compliance export
+    cy.get('input[data-testid="enableComplianceExporttrue"]').click();
+
+    // Change export format to CSV
+    cy.get('select[data-testid="exportFormatdropdown"]').select('csv');
+
+    // Save settings
+    cy.get('button[data-testid="saveSetting"]').click();
+
+    waitUntilConfigSave();
+}
+
+function gotoCompliancePage() {
+    cy.visit('/admin_console/compliance/export');
+}
+
+function gotoTeamAndPostImage() {
+    // Get user teams
+    cy.apiGetTeamsForUser().then(({teams}) => {
+        const team = teams[0];
+        cy.visit(`/${team.name}/channels/town-square`); // Go to the first found team
+    });
```

Comment:
This is needed locally when for some reason the test fails and there are still images attached to the footer.

```suggestion
    });
    
    // # Remove images from post message footer if exist
    cy.waitUntil(() => cy.get('#postCreateFooter').then((el) => {
        if (el.find('.post-image.normal').length > 0) {
            cy.get('.file-preview__remove > .icon').click();
        }
        return el.find('.post-image.normal').length == 0;
    }));
```